File: shopcart/backend/src/main/java/com/ecommerce/service/CartService.java
package com.ecommerce.service;

import com.ecommerce.dto.response.CartResponse;

public interface CartService {
    CartResponse getCart();
    CartResponse addToCart(Long productId, int quantity);
    CartResponse updateCartItem(Long cartItemId, int quantity);
    CartResponse removeCartItem(Long cartItemId);
    void clearCart();
}

File: shopcart/frontend/src/services/cartService.js
import api, { apiMultipart } from './api';

// ── Cart Service ──
const cartService = {
  getCart: () => api.get('/cart'),
  addToCart: (productId, quantity) => api.post('/cart/add', { productId, quantity }),
  updateCartItem: (cartItemId, quantity) => api.put(`/cart/update/${cartItemId}`, { quantity }),
  removeCartItem: (cartItemId) => api.delete(`/cart/remove/${cartItemId}`),
  clearCart: () => api.delete('/cart/clear'),
};
export default cartService;
